cron/build.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+
+# Parameters
+##############################################################################
+
+if [ "$#" -ne 3 ]; then
+    echo 'Illegal number of parameters'
+    echo '     build.sh (conda|manywheel) DESIRED_PYTHON DESIRED_CUDA'
+    echo 'e.g. build.sh manywheel 2.7mu cu80'
+    echo 'e.g. build.sh conda 2.7 cpu'
+    echo ' DESIRED_PYTHON must match:   \d.\d(mu?)?'
+    echo ' DESIRED_CUDA must match  :   (cpu|cu\d\d)'
+    exit 1
+fi
+package_type="$1"
+desired_python="$2"
+desired_cuda="$3"
+
+set -ex
+
+# Validate parameters
+if [[ "$package_type" != 'conda' && "$package_type" != 'manywheel' ]]; then
+    echo "This script doesn't handle packages of type $package_type"
+    exit 1
+fi
+echo "Building a $package_type package for python$desired_python and $desired_cuda"
+echo "Starting a new build at $(date)"
+
+# Move to today's workspace folder
+today="/scratch/nightlies/$(date +%Y_%m_%d)"
+if [[ ! -d "$today" ]]; then
+    echo "The prep job for today's nightlies has not been run correctly"
+    exit 1
+fi
+builder_root_dir="${today}/builder"
+pytorch_root_dir="${today}/pytorch"
+host_package_dir="$today"
+docker_package_dir="/host_machine_pkgs"
+
+# Map cuda/python/storage dirs for conda or manywheel
+python_nodot="${desired_python:0:1}${desired_python:2:1}"
+if [[ "$desired_cuda" == 'cpu' ]]; then
+    build_for_cpu=1
+else
+    cuda_nodot="${desired_cuda:2:2}"
+fi
+if [[ "$package_type" == 'conda' ]]; then
+    desired_python="${desired_python:0:3}"
+    if [[ -n "$build_for_cpu" ]]; then
+        desired_cuda='cpu'
+    else
+        desired_cuda="$cuda_nodot"
+    fi
+    build_script='/remote/conda/build_pytorch.sh'
+    docker_image="soumith/conda-cuda"
+else
+    building_manywheels=1
+    if [[ "$desired_python" == '2.7mu' ]]; then
+        desired_python='cp27-cp27mu'
+    else
+        desired_python="cp${python_nodot}-cp${python_nodot}m"
+    fi
+    # desired_cuda is correct
+    if [[ -n "$build_for_cpu" ]]; then
+        build_script='/remote/manywheel/build_cpu.sh'
+    else
+        build_script='/remote/manywheel/build.sh'
+    fi
+    if [[ -n "$build_for_cpu" ]]; then
+        docker_image="soumith/manylinux-cuda80"
+    else
+        docker_image="soumith/manylinux-cuda$cuda_nodot"
+    fi
+fi
+
+# Set default parameters
+if [[ -z "$TORCH_PACKAGE_NAME" ]]; then
+    # TODO handle pips converting - to _
+    if [[ -n "$building_manywheels" ]]; then
+        TORCH_PACKAGE_NAME='torch_nightly'
+    else
+        TORCH_PACKAGE_NAME='torch-nightly'
+    fi
+fi
+if [[ -z "$PYTORCH_BUILD_VERSION" ]]; then
+    PYTORCH_BUILD_VERSION='nightly'
+fi
+if [[ -z "$PYTORCH_BUILD_NUMBER" ]]; then
+    PYTORCH_BUILD_NUMBER='1'
```

Comment:
Do you want this to be `1.0` instead of 1? Just checking.

Reply:
No the versions of these will be the date followed by '.post$PYTORCH_BUILD_NUMBER' if it's greater than 1. This is so we can have multiple builds per day